test/e2e/ginkgo/multiple_cluster_deployment_test.go

```diff
@@ -0,0 +1,178 @@
+			_, err = ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "nginx", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("deletes WEC objects following deletion of the bindingpolicy object in wds", func() {
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		ginkgo.It("handles multiple bindingpolicies with overlapping matches", func() {
+			ginkgo.By("creates a second bindingpolicy with overlapping matches")
+			util.CreateBindingPolicy(ctx, ksWds, "nginx-2",
+				map[string]string{"location-group": "edge"},
+				map[string]string{"app.kubernetes.io/name": "nginx"})
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+
+			ginkgo.By("delete the second bindingpolicy")
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx-2", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 1)
+		})
+
+		ginkgo.It("deletes WEC objects when wds deployment is deleted", func() {
+			util.DeleteDeployment(ctx, wds, ns, "nginx")
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+		ginkgo.It("deletes WEC objects when BindingPolicy is deleted", func() {
+			err := ksWds.ControlV1alpha1().BindingPolicies().Delete(ctx, "nginx", metav1.DeleteOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 0)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+		})
+
+	})
+
+	ginkgo.Context("singleton status testing", func() {
+		ginkgo.It("sets singleton status when a singleton bindingpolicy/deployment is created", func() {
+			util.DeleteDeployment(ctx, wds, ns, "nginx") // we don't have to delete nginx
+			util.CreateDeployment(ctx, wds, ns, "nginx-singleton", "nginx-singleton")
+			util.CreateBindingPolicy(ctx, ksWds, "nginx-singleton",
+				map[string]string{"name": "cluster1"},
+				map[string]string{"app.kubernetes.io/name": "nginx-singleton"})
+			patch := []byte(`{"spec":{"wantSingletonReportedState": true}}`)
+			_, err := ksWds.ControlV1alpha1().BindingPolicies().Patch(
+				ctx, "nginx-singleton", types.MergePatchType, patch, metav1.PatchOptions{})
+			gomega.ExpectWithOffset(1, err).NotTo(gomega.HaveOccurred())
+			util.ValidateNumDeployments(ctx, wec1, ns, 1)
+			util.ValidateNumDeployments(ctx, wec2, ns, 0)
+			util.ValidateSingletonStatus(ctx, wds, ns, "nginx-singleton")
+		})
+	})
+
+	ginkgo.Context("object field filtering", func() {
```

Comment:
I think we more commonly call this "object cleaning".

Reply:
I changed it to cleaning.

Reply:
https://github.com/effi-ofer/edge-mc/blob/73515206ca859c31aebbf5d56b010dc5e1cfdea7/test/e2e/ginkgo/multiple_cluster_deployment_test.go#L133 still speaks of "field filtering".

Reply:
I just scrabbed the word filtering out.

Reply:
"Field cleaning" is still too specific. An individual label or annotation is not a "field". Just plain "cleaning" works.

Reply:
The word field is removed.